Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al

```diff
@@ -455,4 +455,100 @@ codeunit 4022 "GP Vendor Migrator"
+                            VendorPostingGroup.Validate("Description", GPPM00100.VNDCLDSC);
+
+                            // Payables Account
+                            AccountNumber := GetAccountNumber(GPPM00100.PMAPINDX);
+                            if AccountNumber <> '' then begin
+                                EnsureAccountHasGenProdPostingAccount(AccountNumber);
+                                VendorPostingGroup.Validate("Payables Account", AccountNumber);
+                            end;
+
+                            // Service Charge Acc.
+                            AccountNumber := GetAccountNumber(GPPM00100.PMFINIDX);
+                            if AccountNumber <> '' then begin
+                                EnsureAccountHasGenProdPostingAccount(AccountNumber);
+                                VendorPostingGroup.Validate("Service Charge Acc.", AccountNumber);
+                            end;
+
+                            // Payment Disc. Debit Acc.
+                            AccountNumber := GetAccountNumber(GPPM00100.PMDTKIDX);
+                            if AccountNumber <> '' then begin
+                                EnsureAccountHasGenProdPostingAccount(AccountNumber);
+                                VendorPostingGroup.Validate("Payment Disc. Debit Acc.", AccountNumber);
+                            end;
+
+                            // Payment Disc. Credit Acc.
+                            AccountNumber := GetAccountNumber(GPPM00100.PMDAVIDX);
+                            if AccountNumber <> '' then begin
+                                EnsureAccountHasGenProdPostingAccount(AccountNumber);
+                                VendorPostingGroup.Validate("Payment Disc. Credit Acc.", AccountNumber);
+                            end;
+
+                            // Payment Tolerance Debit Acc.
+                            // Payment Tolerance Credit Acc.
+                            AccountNumber := GetAccountNumber(GPPM00100.PMWRTIDX);
+                            if AccountNumber <> '' then begin
+                                EnsureAccountHasGenProdPostingAccount(AccountNumber);
+                                VendorPostingGroup.Validate("Payment Tolerance Debit Acc.", AccountNumber);
+                                VendorPostingGroup.Validate("Payment Tolerance Credit Acc.", AccountNumber);
+                            end;
+
+                            VendorPostingGroup.Insert();
+
+                            Vendor.Validate("Vendor Posting Group", ClassId);
+                            Vendor.Modify(true);
+                        end;
+        until GPPM00200.Next() = 0;
+    end;
+
+    local procedure GetAccountNumber(GPAccountIndex: Integer): Text[75]
+    var
+        GPAccount: Record "GP Account";
+    begin
+        if (GPAccountIndex > 0) then
+            if GPAccount.Get(GPAccountIndex) then
+                exit(GPAccount.AcctNum);
```

Comment:
Should we be getting Account Number from BC instead of from the staging table? Possible difference in account information due to how the account migration sets the BC value?

Reply:
I agree with using the account index...that is correct. I think there could be issues with the account number though. Just a thought.

Reply:
I didn't notice the field length difference, I'll make that change now.